Problem: java.lang.NoClassDefFoundError: net/sf/cglib/proxy/Callback

error why running JUnit.

Note that I am not using querydsl at all, see code at #56

```
java.lang.NoClassDefFoundError: net/sf/cglib/proxy/Callback
	at com.querydsl.core.alias.Alias.<clinit>(Alias.java:58)
	at com.mmnaseri.utils.spring.data.repository.DefaultQueryDslPredicateExecutor.setDataStore(DefaultQueryDslPredicateExecutor.java:60)
	at com.mmnaseri.utils.spring.data.proxy.impl.DefaultRepositoryFactory.getTypeMappings(DefaultRepositoryFactory.java:137)
	at com.mmnaseri.utils.spring.data.proxy.impl.DefaultRepositoryFactory.getInstance(DefaultRepositoryFactory.java:69)
	at com.mmnaseri.utils.spring.data.dsl.mock.RepositoryMockBuilder.mock(RepositoryMockBuilder.java:103)
	at com.mmnaseri.utils.spring.data.dsl.mock.RepositoryMockBuilder.mock(RepositoryMockBuilder.java:101)
	at com.mmnaseri.utils.spring.data.dsl.factory.RepositoryFactoryBuilder.mock(RepositoryFactoryBuilder.java:273)
	at com.bee2c.platform.MbpDatabaseRelationsTestManual.testDemo(MbpDatabaseRelationsTestManual.java:24)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
Caused by: java.lang.ClassNotFoundException: net.sf.cglib.proxy.Callback
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 31 more


```

and code
```java
import org.junit.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.bee2c.platform.entity.FbpMbpPolicy;
import com.bee2c.platform.entity.FbpMbpPolicyDate;
import com.bee2c.platform.repository.FbpMbpPolicyDateRepository;
import com.bee2c.platform.repository.FbpMbpPolicyRepository;

import com.mmnaseri.utils.spring.data.dsl.factory.RepositoryFactoryBuilder;

public class MbpDatabaseRelationsTestManual {
	
	private static final Logger log = LoggerFactory.getLogger(MbpDatabaseRelationsTestManual.class);


	@Test
	public void testDemo() {
		
		final FbpMbpPolicyRepository repository = RepositoryFactoryBuilder.builder().mock(FbpMbpPolicyRepository.class);
		final FbpMbpPolicyDateRepository policyDateRepository = RepositoryFactoryBuilder.builder().mock(FbpMbpPolicyDateRepository.class);

		FbpMbpPolicy policy = new FbpMbpPolicy();
		repository.save(policy);
		log.info(policy.toString());
		
		FbpMbpPolicyDate policyDate = new FbpMbpPolicyDate();
		policyDate.setMbpPolicy(policy);
		policyDateRepository.save(policyDate);
		log.info(policyDate.toString());

		FbpMbpPolicyDate policyDate2 = new FbpMbpPolicyDate();
		policyDate2.setMbpPolicy(policy);
		policyDateRepository.save(policyDate2);
		log.info(policyDate2.toString());
		
		
	}
```

--- a/spring-data-mock/src/main/java/com/mmnaseri/utils/spring/data/proxy/impl/DefaultTypeMappingContext.java
+++ b/spring-data-mock/src/main/java/com/mmnaseri/utils/spring/data/proxy/impl/DefaultTypeMappingContext.java
@@ -27,6 +27,11 @@ import java.util.concurrent.ConcurrentMap;
 public class DefaultTypeMappingContext implements TypeMappingContext {
 
     private static final Log log = LogFactory.getLog(DefaultTypeMappingContext.class);
+    private static final String GEMFIRE_SUPPORT_CLASS = "org.springframework.data.gemfire.repository.GemfireRepository";
+    private static final String JPA_SUPPORT_CLASS = "org.springframework.data.jpa.repository.JpaRepository";
+    private static final String QUERYDSL_SUPPORT_CLASS = "org.springframework.data.querydsl.QueryDslPredicateExecutor";
+    private static final String QUERY_BY_EXAMPLE_SUPPORT_CLASS = "org.springframework.data.repository.query.QueryByExampleExecutor";
+    private static final String CGLIB_SUPPORT_CLASS = "net.sf.cglib.proxy.Callback";
     private final TypeMappingContext parent;
     private ConcurrentMap<Class<?>, List<Class<?>>> mappings = new ConcurrentHashMap<>();
 
@@ -45,26 +50,29 @@ public class DefaultTypeMappingContext implements TypeMappingContext {
         this(null);
         if (registerDefaults) {
             log.info("Trying to register all the default type mappings");
-            final ClassLoader defaultClassLoader = ClassUtils.getDefaultClassLoader();
-            if (ClassUtils.isPresent("org.springframework.data.gemfire.repository.GemfireRepository", defaultClassLoader)) {
+            if (isClassPresent(GEMFIRE_SUPPORT_CLASS)) {
                 log.debug("We seem to have Gemfire in the classpath, so, we should register the supporting registry");
                 register(Object.class, DefaultGemfireRepository.class);
             }
-            if (ClassUtils.isPresent("org.springframework.data.jpa.repository.JpaRepository", defaultClassLoader)) {
+            if (isClassPresent(JPA_SUPPORT_CLASS)) {
                 log.debug("JPA support is enabled in this project, so we need to support the methods");
                 register(Object.class, DefaultJpaRepository.class);
             }
-            if (ClassUtils.isPresent("org.springframework.data.querydsl.QueryDslPredicateExecutor", defaultClassLoader)) {
+            if (isClassPresent(QUERYDSL_SUPPORT_CLASS) && isClassPresent(CGLIB_SUPPORT_CLASS)) {
                 log.debug("QueryDSL support is enabled. We will add the proper method implementations.");
                 register(Object.class, DefaultQueryDslPredicateExecutor.class);
             }
-            if (ClassUtils.isPresent("org.springframework.data.repository.query.QueryByExampleExecutor", defaultClassLoader)) {
+            if (isClassPresent(QUERY_BY_EXAMPLE_SUPPORT_CLASS)) {
                 log.debug("Query by example is enabled. We will the proper method implementations.");
                 register(Object.class, DefaultQueryByExampleExecutor.class);
             }
             register(Object.class, DefaultPagingAndSortingRepository.class);
             register(Object.class, DefaultCrudRepository.class);
         }
+    }
+
+    private boolean isClassPresent(String className) {
+        return ClassUtils.isPresent(className, ClassUtils.getDefaultClassLoader());
     }
 
     public DefaultTypeMappingContext(TypeMappingContext parent) {

